src/webviews/extension-side/dataviewer/dataViewerDependencyService.ts

```diff
@@ -0,0 +1,123 @@
+ */
+@injectable()
+export class DataViewerDependencyService implements IDataViewerDependencyService {
+    constructor(
+        @inject(IApplicationShell) private readonly applicationShell: IApplicationShell,
+        @inject(IsCodeSpace) private isCodeSpace: boolean
+    ) {}
+
+    private packaging(kernel: IKernel): 'pip' | 'conda' {
+        const envType = kernel.kernelConnectionMetadata.interpreter?.envType;
+        const isConda = envType === EnvironmentType.Conda;
+        return isConda ? 'conda' : 'pip';
+    }
+
+    public async checkAndInstallMissingDependenciesOnEnvironment(): Promise<void> {
+        throw new Error('Not implemented');
+    }
+
+    public async checkAndInstallMissingDependenciesOnKernel(kernel: IKernel): Promise<void> {
+        const pandasVersion = await this.getVersionOfPandas(kernel);
+
+        if (pandasVersion) {
+            if (isVersionOfPandaSupported(pandasVersion)) {
+                return;
+            }
+            sendTelemetryEvent(Telemetry.PandasTooOld);
+            // Warn user that we cannot start because pandas is too old.
+            const versionStr = `${pandasVersion.major}.${pandasVersion.minor}.${pandasVersion.build}`;
+            throw new Error(DataScience.pandasTooOldForViewingFormat().format(versionStr));
+        }
+
+        sendTelemetryEvent(Telemetry.PandasNotInstalled);
+        await this.installMissingDependencies(kernel);
+    }
+
+    private async installMissingDependencies(kernel: IKernel): Promise<void> {
+        if (!kernel.session) {
+            sendTelemetryEvent(Telemetry.NoActiveKernelSession);
+            throw new Error(DataScience.noActiveKernelSession());
+        }
+
+        sendTelemetryEvent(Telemetry.PythonModuleInstall, undefined, {
+            action: 'displayed',
+            moduleName: ProductNames.get(Product.pandas)!
+        });
+
+        const selection = this.isCodeSpace
+            ? Common.install()
+            : await this.applicationShell.showErrorMessage(
+                  DataScience.pandasRequiredForViewing(),
+                  { modal: true },
+                  Common.install()
+              );
+
+        const command = `${this.packaging(kernel)} install pandas`;
+
+        if (selection === Common.install()) {
+            try {
+                await this.executeSilently(command, kernel);
+                sendTelemetryEvent(Telemetry.UserInstalledPandas);
+            } catch (e) {
+                console.log('Error installing pandas', e);
+                sendTelemetryEvent(Telemetry.FailedToInstallPandas);
+                throw new Error(DataScience.failedToInstallPandas());
+            }
+        } else {
+            sendTelemetryEvent(Telemetry.UserDidNotInstallPandas);
+            throw new Error(DataScience.pandasRequiredForViewing());
+        }
+    }
+
+    private async getVersionOfPandas(kernel: IKernel): Promise<SemVer | undefined> {
+        const command = 'import pandas;print(pandas.__version__)';
```

Comment:
I think we should change this to:
```python
import pandas as _VSCODE_pandas
print(_VSCODE_pandas.__version__)
del _VSCODE_pandas
```

This way we don't change the imported modules in users code.
I.e. if user didn't import pandas we shoudln't leave this module imported into the global (current) scope

Reply:
Updated!